File: src/main/java/org/usfirst/frc/team6325/robot/commands/Drive/ProfileFollower.java
package org.usfirst.frc.team6325.robot.commands.Drive;

import org.usfirst.frc.team6325.robot.Robot;
import edu.wpi.first.wpilibj.command.Command;
import jaci.pathfinder.followers.EncoderFollower;


public class ProfileFollower extends Command 
{
	String leftCSV, rightCSV;
	EncoderFollower[] followers;

	public ProfileFollower(String leftCSV, String rightCSV) 
	{
		leftCSV = this.leftCSV;
		rightCSV = this.rightCSV;
		setInterruptible(false);
		followers = Robot.drivetrain.initPath(leftCSV, rightCSV);
	}

	@Override
	protected void initialize() 
	{
		Robot.drivetrain.executePath(followers, false);
	}

	@Override
	protected void execute() 
	{
		Robot.drivetrain.executePath(followers, false);
	}

	@Override
	protected void end() 
	{
        Robot.drivetrain.drive(0, 0);
    }
	
	@Override
	protected void interrupted() 
	{
	    end();
	}

	@Override
	protected boolean isFinished() 
	{
		return Robot.drivetrain.getIsProfileFinished();
	}
}

File: src/main/java/org/usfirst/frc/team6325/robot/commands/Drive/ProfileFollower.kt
package org.usfirst.frc.team6325.robot.commands.Drive

import org.usfirst.frc.team6325.robot.Polybius
import org.sertain.command.Command
import jaci.pathfinder.followers.EncoderFollower


public class ProfileFollower : Command
{
    var leftCSV: String = null
    var rightCSV: String = null
    var followers[]: EncoderFollower[] = null

    public fun ProfileFollower(leftCSV: String, rightCSV: String)
    {
        leftCSV = this.leftCSV
        rightCSV = this.rightCSV
        setInterruptible(false)
        followers = Polybius.drivetrain.initPath(leftCSV, rightCSV)
    }

    init { Polybius.drivetrain.executePath(followers, false) }

    override fun execute() = Polybius.drivetrain.executePath(followers, true)
    override fun onDestroy() = Polybius.drivetrain.drive(0, 0)
    override fun isCanceled() = onDestroy()
    override fun isCompleted() = return Polybius.drivetrain.getIsProfileFinished()
}